itou/scripts/management/commands/export_auto_prescriptions.py

```diff
@@ -0,0 +1,104 @@
+import csv
+import datetime
+import os
+import sys
+
+from django.core.management.base import BaseCommand
+from django.core.paginator import Paginator
+from django.db.models import F
+
+from itou.common_apps.address.departments import DEPARTMENTS
+from itou.job_applications.models import JobApplication, JobApplicationWorkflow
+from itou.siaes.enums import SiaeKind
+from itou.users.enums import KIND_SIAE_STAFF
+
+
+# this is arbitrary, it works well and gives a manageable number of objects on stdout
+# and in memory. Modify as needed if you wish so.
+DEFAULT_PAGINATION_SIZE = 1000
+
+
+def paginate_queryset(queryset, debug=False):
+    def debug_log(log):
+        if debug:
+            sys.stderr.write(log)
+
+    debug_log(f"paginated_command: queryset objects count={queryset.count()}")
+    # this is the only efficient way I found to both:
+    # - iterate over the objects and treat them on a fire-and-forget basis, without bulk updates.
+    # - iterator() and direct filter() cause OOM (out-of-memory) errors.
+    paginator = Paginator(queryset, DEFAULT_PAGINATION_SIZE)
+    try:
+        debug_log(f"paginated_command total_range={paginator.page_range}")
+        for page_number in paginator.page_range:
+            debug_log(f"paginated_command page number={page_number}")
+            page = paginator.page(page_number)
+            for obj in page.object_list:
+                yield obj
+    except KeyboardInterrupt:
+        debug_log("! keyboard interrupt, exiting.")
+
+
+class Command(BaseCommand):
+    def add_arguments(self, parser):
+        parser.add_argument("--debug", dest="debug", action="store_true", help="Print debug statements")
+
+    def handle(self, debug=False, **kwargs):
+        writer = csv.writer(self.stdout, lineterminator=os.linesep)
+        queryset = (
+            JobApplication.objects.exclude(approval=None)
+            .select_related(
+                "approval",
+                "job_seeker",
+                "to_siae",
+                "to_siae__convention",
+            )
+            .filter(
+                state=JobApplicationWorkflow.STATE_ACCEPTED,
+                to_siae__kind__in=[SiaeKind.ACI, SiaeKind.EI, SiaeKind.ETTI, SiaeKind.AI],
+                eligibility_diagnosis__author_kind=KIND_SIAE_STAFF,
+                eligibility_diagnosis__author_siae=F("to_siae"),
+                approval__start_at__range=[datetime.date(2022, 1, 1), datetime.date(2022, 10, 1)],
+            )
+        )
+        if queryset.count() > 1:
```

Comment:
S’il n’y a qu’un seul résultat, pas de headers ?
Appeler `__bool__` sur le qs permet d’évaluer une seule fois les résultats, au lieu de deux requêtes (SELECT COUNT + SELECT). 
```suggestion
        if queryset:
```

Reply:
Ah ben oui.

Reply:
Heu, en revanche, ça charge tous les résultats en mémoire, ce qui est incompatible avec le `.iterator()` suggéré après.

Reply:
Je conserve pour l'instant du coup.

Reply:
Ou alors, on écrit toujours les headers ?

Reply:
Certes, osef.